src/release-notify-action.js

```diff
@@ -26,13 +31,19 @@ async function runAction(token, notifyAfter, commitMessageLines) {
 
   const notifyDate = notifyAfterToMs(notifyAfter)
 
-  const snoozed = await isSnoozed(token, latestRelease.published_at, notifyDate)
+  const pendingIssue = await getLastOpenPendingIssue(token)
 
-  if (snoozed) {
-    return logInfo('Release notify has been snoozed')
-  }
+  if (!pendingIssue && !ignoreSnoozed) {
```

Comment:
`isSnoozed` only checks if there is a closed (not planned) `notify-release` issue that is not stale, it's now possible if you run a `workflow_dispatch` to have an open `notify-release` issue and a snoozed `notify-release` issue at the same time. That wasn't possible before, because the snoozed issue was always checked first before creating a new issue. This pending (open) check is an alternative to marking the snoozed issue as fully closed.

Reply:
ok understood. with this in mind, do you think we need other changes to the action then to account for the occurrence that there are possibly both a snoozed closed issue and an open pending issue?

Reply:
If there is an open issue, then that should always be updated, ignoring any snoozed issues. The tests should cover the following:

```mermaid
flowchart TD
    pending-issue{Open issue?} -->|Yes| unreleased-commits{Unreleased commits?}
    pending-issue -->|No| ignore-snoozed{Ignore snoozed?}
    ignore-snoozed -->|Yes| unreleased-commits
    ignore-snoozed -->|No| is-snoozed{Is snoozed?}
    is-snoozed -->|Yes| no-action(No action)
    is-snoozed -->|No| unreleased-commits
    unreleased-commits -->|Yes| create-or-update-issue(Create or update issue)
    unreleased-commits -->|No| release-since-issue{Release since issue?}
    release-since-issue -->|Yes| close-issue(Close issue)
    release-since-issue -->|No| no-action
```